test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go

```diff
@@ -550,22 +554,23 @@ func setup(ctx context.Context) func(ctx2 context.Context) {
 			By(fmt.Sprintf("skipping teardown - no changes to profile %q were applied", profile.Name))
 		}
 	}
-	testprofiles.UpdateWithRetry(profile)
-	mcp, err := mcps.GetByProfile(profile)
-	Expect(err).ToNot(HaveOccurred())
+	profiles.UpdateWithRetry(profile)
+	poolName := poolname.GetByProfile(context.TODO(), profile)
+	By(fmt.Sprintf("Applying changes in performance profile and waiting until %s will start updating", poolName))
+	profilesupdate.WaitForTuningUpdating(context.TODO(), profile)
 
-	mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
-	mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+	By(fmt.Sprintf("Waiting when %s finishes updates", poolName))
+	profilesupdate.WaitForTuningUpdated(context.TODO(), profile)
 
 	teardown := func(ctx2 context.Context) {
 		By(fmt.Sprintf("executing teardown - revert profile %q back to its initial state", profile.Name))
-		Expect(testclient.Client.Get(ctx2, client.ObjectKeyFromObject(initialProfile), profile))
-		testprofiles.UpdateWithRetry(initialProfile)
+		Expect(testclient.ControlPlaneClient.Get(ctx2, client.ObjectKeyFromObject(initialProfile), profile))
+		profiles.UpdateWithRetry(initialProfile)
 
 		// do not wait if nothing has changed
 		if initialProfile.ResourceVersion != profile.ResourceVersion {
-			mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
-			mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+			mcps.WaitForCondition(poolName, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
```

Comment:
if this flow will run on HyperShift it will failed.

replace:
```
mcps.WaitForCondition(poolName, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
mcps.WaitForCondition(poolName, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
```
with:
```
profilesupdate.WaitForTuningUpdating(context.TODO(), profile)
profilesupdate.WaitForTuningUpdated(context.TODO(), profile)
```